pyccel/codegen/printing/ccode.py

```diff
@@ -2272,6 +2275,15 @@ def _print_Program(self, expr):
     def _print_CustomDataType(self, expr):
         return "struct " + expr.name
 
+    def _print_Del(self, expr):
+        for variable in expr.variables:
+            if isinstance(variable.dtype, CustomDataType) and not variable.cls_base.is_deallocated:
+                variable_address = self._print(ObjectAddress(variable))
+                Pyccel__del = next(method.name for method in variable.cls_base.methods if method.name.endswith('__del__'))
+                variable.cls_base.is_deallocated = True
+                return(f"{Pyccel__del}({variable_address});\n")
```

Comment:
The return is too early. What if there are multiple class objects being deleted?

Reply:
Fixed 0dbb55d7043fb5c0772d201935163c592e7097eb.